main/src/ca/uwaterloo/flix/language/phase/Lowering.scala

```diff
@@ -1641,6 +1648,31 @@ object Lowering {
     LoweredAst.Expression.Cast(blockExp, None, Some(Type.Pure), Some(Type.Empty), tpe, pur, eff, loc.asSynthetic)
   }
 
+  /**
+    * Returns the expression of a `ParYield` expression that should be evaluated in the calling thread.
+    */
+  private def mkParYieldMainThread(exps: List[LoweredAst.ParYieldFragment], waitExps: LoweredAst.Expression): LoweredAst.Expression = {
```

Comment:
Not "Main" thread, maybe current thread?

Reply:
Calling thread?

Reply:
CurrentThread? :)